### PR TITLE
[htmlparser2] [sanitize-html] [react-html-parser] Deprecated htmlparser2 and updated dependent libraries

### PR DESCRIPTION
**Pull request by daviduzumeri**
Please fill in this template.

- [X] Use a meaningful title for the pull request. Include the name of the package modified.
- [X] Test the change in your own code. (Compile and run.)
- [X] Add or edit tests to reflect the change. (Run with `npm test`.)
- [X] Follow the advice from the [readme](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#make-a-pull-request).
- [X] Avoid [common mistakes](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#common-mistakes).
- [X] Run `npm run lint package-name` (or `tsc` if no `tslint.json` is present).

Select one of these and delete the others:

If removing a declaration:
- [ ] If a package was never on DefinitelyTyped, you don't need to do anything. (If you wrote a package and provided types, you don't need to register it with us.)
- [X] Delete the package's directory.
- [X] Add it to `notNeededPackages.json`.

I'll set up a PR to the whitelist for the main package. `sanitize-html` I was able to test locally but I could use a hand with `react-html-parser`. Due to a bunch of downstream changes involving `domhandler` and `domutils` this whole thing was pretty tricky, but the existing types package (for sanitize-html, at least) just isn't working at all since it keeps grabbing the new built-in htmlparser2 types, which are incompatible.

### CODE REVIEW

File: types/sanitize-html/index.d.ts [hunk 1]
```diff
@@ -12,64 +12,56 @@
 // Definitions: https://github.com/DefinitelyTyped/DefinitelyTyped
 // TypeScript Version: 2.2
 
-///<reference types="htmlparser2"/>
-
-import { Options } from "htmlparser2";
+import { ParserOptions, DomHandlerOptions } from 'htmlparser2';
```
Comment (WillGibson) [inline types/sanitize-html/index.d.ts]: `ParserOptions` and `DomHandlerOptions` do not appear to be used in the code below, whereas the deleted `Options` is.

File: notNeededPackages.json [hunk 2]
```diff
@@ -5016,18 +5022,18 @@
...
             "typingsPackageName": "theme-ui__color",
             "sourceRepoURL": "https://github.com/system-ui/theme-ui",
             "asOfVersion": "0.3.1"
         },
+        {
+            "libraryName": "theming",
+            "typingsPackageName": "theming",
+            "sourceRepoURL": "https://github.com/cssinjs/theming",
+            "asOfVersion": "2.0.0"
+        },
```
Comment (WillGibson) [inline notNeededPackages.json]: 👍 for the alphabetising :-)

File: types/sanitize-html/index.d.ts [hunk 3]
```diff
@@ -12,64 +12,56 @@
...
+        exclusiveFilter?: (frame: IFrame) => boolean;
+        nonTextTags?: string[];
+        selfClosing?: string[];
+        transformTags?: { [tagName: string]: string | Transformer };
+        parser?: Options;
+    }
+
+    var defaults: IDefaults;
+
+    function simpleTransform(tagName: string, attribs: Attributes, merge?: boolean): Transformer;
```
Comment (WillGibson) [inline types/sanitize-html/index.d.ts]: I'm all for the indenting tidy up, but it makes it very hard to see what actually changed, if anything, in here. It's easier for reviewers if you submit formatting changes in a separate PR, then make the actual changes after. I know it's hard to keep to that though, I use Intellij's Reformat Code all the time, and sometimes on an older file it does a lot of work!

For the purpose of this PR, please could you add comments highlighting any actual changes in there.

**Review by WillGibson — COMMENTED** [review]
Thanks for taking the  time to do this work @daviduzumeri .

File: types/sanitize-html/index.d.ts (hunk 1 above)
Comment (daviduzumeri) [reply types/sanitize-html/index.d.ts]: Fixed! Thanks for the catch, I had changed it in the local version I was testing with but forgot to copy that line over. :( With whitelist not updated yet and already getting TSC errors due to htmlparser2 not being node_modules my editor isn't catching my ass on this

File: types/sanitize-html/index.d.ts (hunk 3 above)
Comment (daviduzumeri) [reply types/sanitize-html/index.d.ts]: Undid all the formatting changes; good call.

**Comment by WillGibson** [comment]
This looks good to me now @daviduzumeri, but the CI builds need to be passing.

**Comment by daviduzumeri** [comment]
it seems to be the dependency chain continuing with another deprecated types package that’s still on here, I’ll take a look this weekend

**Comment by daviduzumeri** [comment]
Okay! It looks like nothing else was using `domhandler` or `domutils` outside of `htmlparser2`'s native typings, so crossing fingers this passes and then we can call this a day.

**Comment by daviduzumeri** [comment]
Okay, so it looks like `tsc` passes when I use this definition file locally, but it won't pass here because `domhandler` and `domutils` include their TypeScript typings via including the `src` directory rather than generating a `.d.ts` file and properly referencing it in `package.json`, which is placing me at ... a bit of a loss as to how to proceed.

**Pull request closed without merging**